src/main/java/request/HttpRequestBuilder.java

```diff
@@ -1,41 +1,27 @@
 package com.td.HttpServer;
 
-import java.util.*;
+import java.util.HashMap;
 import java.io.IOException;
 
 public class HttpRequestBuilder implements IRequestBuilder {
+  IRequestReader reader;
+  IRequestParser parser;
 
-  private IValidator requestVerifier;
-  public HttpRequestBuilder(IValidator requestVerifier) {
-    this.requestVerifier = requestVerifier;
+  public HttpRequestBuilder(IRequestReader reader, IRequestParser parser ) {
+    this.reader = reader;
+    this.parser = parser;
   }
 
-  public HttpRequest createRequest(String rawRequest) throws InvalidHttpRequestException {
+  public HttpRequest createRequest(IMessageIO client) throws InvalidHttpRequestException, IOException {
```

Comment:
This is really more like `getNextRequest`, right? It doesn't just build it out of nothing.